hoomd/md/external/wall.py

```diff
@@ -0,0 +1,577 @@
+        provides features and documentation common to all standard wall
+        potentials.
+
+    All wall potential classes specify that a given potential energy and
+    force be computed on all particles in the system when the signed cutoff
+    distance is near the wall surface: :math:`r < r_{\mathrm{cut}}`.
+    The force :math:`\vec{F}` is in the direction of :math:`\vec{r}`, the vector
+    pointing from the particle to closest point on the wall's surface and
+    :math:`V_{\mathrm{pair}}(r)` is the pair potential specified by subclasses
+    of `WallPotential`.  Walls are two-sided surfaces with positive signed
+    distances to points on the active side of the wall and negative signed
+    distances to points on the inactive side. Additionally, the wall's mode
+    controls how forces and energies are computed for particles on or near the
+    inactive side. The ``inside`` flag (or ``normal`` in the case of
+    `hoomd.wall.Plane`) determines which side of the surface is active.
+
+    .. rubric:: Standard Mode.
+
+    In the standard mode, when :math:`r_{\mathrm{extrap}} \le 0`, the potential
+    energy is only computed on the active side. :math:`V(r)` is evaluated in the
+    same manner as when the mode is shift for the analogous :py:mod:`pair
+    <hoomd.md.pair>` potentials within the boundaries of the active space.
+
+    .. math::
+
+        V(r)  = V_{\mathrm{pair}}(r) - V_{\mathrm{pair}}(r_{\mathrm{cut}})
+
+    For ``open=True`` spaces:
+
+    .. math::
+        :nowrap:
+
+        \begin{eqnarray*}
+        \vec{F}  = & -\nabla V(r) & 0 < r < r_{\mathrm{cut}} \\
+                 = & 0 & r \ge r_{\mathrm{cut}} \\
+                 = & 0 & r \le 0
+        \end{eqnarray*}
+
+    For ``open=False`` (closed) spaces:
+
+    .. math::
+        :nowrap:
+
+        \begin{eqnarray*}
+        \vec{F}  = & -\nabla V(r) & 0 \le r < r_{\mathrm{cut}} \\
+                 = & 0 & r \ge r_{\mathrm{cut}} \\
+                 = & 0 & r < 0
+        \end{eqnarray*}
+
+    Below we show the potential for a `hoomd.wall.Sphere` with radius 5 in 2D,
+    using the Gaussian potential with :math:`\epsilon=1, \sigma=1` and
+    ``inside=True``.
+
+    .. image:: md-wall-potential.svg
+
+    When ``inside=False`` the potential becomes,
+
+    .. image:: md-wall-potential-outside.svg
+
+    .. rubric: Extrapolated Mode:
+
+    The wall potential can be linearly extrapolated starting at
+    :math:`r = r_{\mathrm{extrap}}` on the active side and continuing to the
+    inactive side. This can be useful to move particles from the inactive side
+    to the active side.
+
+    The extrapolated potential has the following form:
+
+    .. math::
+        :nowrap:
+
+        \begin{eqnarray*}
+        V_{\mathrm{extrap}}(r) =& V(r) &, r > r_{\rm extrap} \\
+             =& V(r_{\rm extrap})
+                + (r_{\rm extrap}-r)\vec{F}(r_{\rm extrap})
+                \cdot \vec{n}&, r \le r_{\rm extrap}
```

Comment:
```suggestion
                \cdot \vec{n}\,\,\,& r \le r_{\rm extrap}
```

Same for the equations below.